droidlet/lowlevel/robot_mover_utils.py

```diff
@@ -110,3 +117,74 @@ def xyz_pyrobot_to_canonical_coords(xyz):
 def xyz_canonical_coords_to_pyrobot_coords(xyz):
     """converts 3D coords from canonical to pyrobot coords."""
     return xyz @ np.linalg.inv(pyrobot_to_canonical_frame)
+
+
+class TrajectoryDataSaver:
+    def __init__(self, root):
+        print(f'TrajectoryDataSaver saving to {root}')
+        self.save_folder = root
+        self.img_folder = os.path.join(self.save_folder, "rgb")
+        self.img_folder_dbg = os.path.join(self.save_folder, "rgb_dbg")
+        self.depth_folder = os.path.join(self.save_folder, "depth")
+        self.seg_folder = os.path.join(self.save_folder, "seg")
+        self.trav_folder = os.path.join(self.save_folder, "trav")
+
+        if os.path.exists(self.save_folder):
+            print(f'rmtree {self.save_folder}')
+            shutil.rmtree(self.save_folder)
+
+        print(f'trying to create {self.save_folder}')
+        Path(self.save_folder).mkdir(parents=True, exist_ok=True)
+
+        for x in [self.img_folder, self.img_folder_dbg, self.depth_folder, self.seg_folder, self.trav_folder]:
+            self.create(x)
+
+        self.pose_dict = {}
+        self.img_count = 0
+        self.dbg_str = "None"
+
+    def create(self, d):
+        if not os.path.isdir(d):
+            os.makedirs(d)
+
+    def set_dbg_str(self, x):
+        self.dbg_str = x
+    
+    def get_total_frames(self):
+        return self.img_count
+
+    def save(self, rgb, depth, seg, pos):
```

Comment:
could you put a comment with format of rgb, depth, seg, and pos?

Reply:
note to self: add comments, add tests